train.py

```diff
@@ -98,50 +101,93 @@ def train(params):
     # Load hgn from parameters to deice
     hgn = load_hgn(params=params, device=device, dtype=dtype)
 
-    # Dataloader
-    dataset_len = params["optimization"]["epochs"] * params["optimization"][
-        "batch_size"]
-    seed = None if params["dataset"]["random"] else 0
-    trainDS = EnvironmentSampler(
-        environment=env,
-        dataset_len=dataset_len,
-        number_of_frames=params["rollout"]["seq_length"],
-        delta_time=params["rollout"]["delta_time"],
-        number_of_rollouts=params["optimization"]["batch_size"],
-        img_size=params["dataset"]["img_size"],
-        color=params["rollout"]["n_channels"] == 3,
-        noise_std=params["dataset"]["noise_std"],
-        radius_bound=params["dataset"]["radius_bound"],
-        world_size=params["dataset"]["world_size"],
-        seed=seed,
-        dtype=dtype)
-    # Dataloader instance test, batch_mode disabled
-    data_loader = torch.utils.data.DataLoader(trainDS,
-                                              shuffle=False,
-                                              batch_size=None)
+    # Either generate data on-the-fly or load the data from disk
+    data_loader = None
+    if params["dataset"]["on_the_fly_data"]:
+        dataset_len = params["optimization"]["epochs"] * params[
+            "optimization"]["batch_size"]
+        trainDS = EnvironmentSampler(
+            environment=env,
+            dataset_len=dataset_len,
+            number_of_frames=params["rollout"]["seq_length"],
+            delta_time=params["rollout"]["delta_time"],
+            number_of_rollouts=params["optimization"]["batch_size"],
+            img_size=params["dataset"]["img_size"],
+            color=params["rollout"]["n_channels"] == 3,
+            noise_std=params["dataset"]["noise_std"],
+            radius_bound=params["dataset"]["radius_bound"],
+            world_size=params["dataset"]["world_size"],
+            seed=None)
+
+        data_loader = torch.utils.data.DataLoader(trainDS,
+                                                  shuffle=False,
+                                                  batch_size=None)
```

Comment:
This is a weird behavior. DataLoader expects this to be an `int`, but if I put `batch_size=1` the returned rollouts have an additional dimension. This is because `EnvironmentSampler` should return single rollouts in `__get_item__()` but it returns batches instead. Therefore, `DataLoader` adds a dimension corresponding to the batch size. 
I don't know why it works with `batch_size=None`, but it shouldn't according to [the docs](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader). 

We should make `EnvironmentSampler` return single samples so that we can use the `batch_size` parameter of the data loader

Reply:
Ok this was not a problem, `DataLoader` can accept `batch_size=Nose` and in that case it won't produce the additional dimension at all.